ihmc-high-level-behaviors/src/main/java/us/ihmc/perception/RapidHeightMapManager.java

```diff
@@ -27,101 +28,97 @@
-      cameraPoseForHeightMap.changeFrame(ReferenceFrame.getWorldFrame());
-
-      rapidHeightMapExtractor.update(sensorToWorldForHeightMap, sensorToGroundForHeightMap, groundToWorldForHeightMap);
-
-      Mat croppedHeightMapImage = rapidHeightMapExtractor.getTerrainMapData().getHeightMap();
-
-      OpenCVTools.compressImagePNG(croppedHeightMapImage, compressedCroppedHeightMapPointer);
-      PerceptionMessageTools.publishCompressedDepthImage(compressedCroppedHeightMapPointer,
-                                                         PerceptionAPI.HEIGHT_MAP_CROPPED,
-                                                         croppedHeightMapImageMessage,
-                                                         ros2,
-                                                         cameraPoseForHeightMap,
-                                                         imageAquisitionTime,
-                                                         rapidHeightMapExtractor.getSequenceNumber(),
-                                                         croppedHeightMapImage.rows(),
-                                                         croppedHeightMapImage.cols(),
-                                                         (float) RapidHeightMapExtractor.getHeightMapParameters().getHeightScaleFactor());
-   }
-
-   public HeightMapData getLatestHeightMapData()
-   {
-      HeightMapData temp = new HeightMapData((float) RapidHeightMapExtractor.getHeightMapParameters().getGlobalCellSizeInMeters(),
-                                             (float) RapidHeightMapExtractor.getHeightMapParameters().getGlobalWidthInMeters(),
-                                             rapidHeightMapExtractor.getSensorOrigin().getX(),
-                                             rapidHeightMapExtractor.getSensorOrigin().getY());
-      RapidHeightMapExtractor.packHeightMapData(rapidHeightMapExtractor, temp);
-      return temp;
-   }
-
-   public TerrainMapData getTerrainMapData()
-   {
-      return rapidHeightMapExtractor.getTerrainMapData();
-   }
-
-   public void destroy()
-   {
-      rapidHeightMapExtractor.destroy();
-   }
-}
+public class RapidHeightMapManager {
+    private final RapidHeightMapExtractorCuda rapidHeightMapExtractor;
+    private final ImageMessage croppedHeightMapImageMessage = new ImageMessage();
+    private final FramePose3D cameraPoseForHeightMap = new FramePose3D();
+    private final RigidBodyTransform sensorToWorldForHeightMap = new RigidBodyTransform();
+    private final RigidBodyTransform sensorToGroundForHeightMap = new RigidBodyTransform();
+    private final RigidBodyTransform groundToWorldForHeightMap = new RigidBodyTransform();
+    private final GpuMat deviceDepthImage;
+    private final Mat hostDepthImage;
+    private final Notification resetHeightMapRequested = new Notification();
+    private final BytePointer compressedCroppedHeightMapPointer = new BytePointer();
+
+    public RapidHeightMapManager(OpenCLManager openCLManager,
```

Comment:
We can remove `openCLManager` from the parameters, right?

Reply:
Other Instaces of the class use it, ill have to change it in otherplaces.